server/paymentProviders/thegivingblock/index.js

```diff
@@ -109,11 +112,15 @@ export const processOrder = async order => {
   await refresh(account);
 
   // create wallet address
-  const { depositAddress, pledgeId } = await createDepositAddress(account, {
+  const { depositAddress, pledgeId, errorMessage, meta } = await createDepositAddress(account, {
     organizationId: account.data.organizationId,
     pledgeAmount: order.data.customData.pledgeAmount,
     pledgeCurrency: order.data.customData.pledgeCurrency,
   });
+  if (errorMessage) {
+    logger.error(`The Giving Block: ${errorMessage} ${meta.errorCode}`);
+    throw new Error(GENERIC_ERROR_MSG);
+  }
```

Comment:
Why is this needed if we have `handleErrorsAndRetry`?

Add error handling there and refactor to avoid duplicate code.

Reply:
Yes, I've refactored code to handle the error within `handleErrorsAndRetry`. However it seems that there's cases where we could first have the `INVALID_JWT_TOKEN` error and then when we try to make the request again within the `handleErrorsAndRetry` then we could silently fail as we don't handle the response error there. I've handled this case as well. I could reproduce this in my local. 👍🏽 

Let me know how it looks now. 🤔